src/Features/LanguageServer/Protocol/Handler/AbstractRequestHandler.cs

```diff
@@ -14,6 +14,8 @@ internal abstract class AbstractRequestHandler<RequestType, ResponseType> : IReq
     {
         protected readonly ILspSolutionProvider SolutionProvider;
 
+        public virtual RequestProcessingMode Type => RequestProcessingMode.Parallel;
```

Comment:
probably better for this to be astract.  so that everyone has to think about what behavior they have.